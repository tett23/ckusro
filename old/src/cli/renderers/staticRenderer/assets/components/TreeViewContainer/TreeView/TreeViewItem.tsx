import React from 'react';
import {
  FileBuffer,
  fileBufferName,
} from '../../../../../../../models/FileBuffer';

export type Props = {
  fileBuffer: FileBuffer;
};

export default function TreeViewItem({ fileBuffer }: Props) {
  return (
    <li>
      <div>{fileBufferName(fileBuffer)}</div>
    </li>
  );
}
