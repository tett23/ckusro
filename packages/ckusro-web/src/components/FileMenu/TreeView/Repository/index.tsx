import { CommitObject, RepositoryInfo } from '@ckusro/ckusro-core';
import React from 'react';
import Cloned from './Cloned';
import HaveNotBeenCloned from './HaveNotBeenCloned';

export type RepositoryProps = {
  repository: RepositoryInfo;
  commitObject: CommitObject | null;
};

export default function Repository(props: RepositoryProps) {
  const { repository, commitObject } = props;
  if (commitObject == null) {
    return <HaveNotBeenCloned {...props} />;
  }

  return <Cloned repository={repository} commitObject={commitObject} />;
}
