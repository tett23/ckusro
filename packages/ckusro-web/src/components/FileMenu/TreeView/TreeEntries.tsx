import {
  TreeEntry as TreeEntryType,
  InternalPath,
  createInternalPath,
} from '@ckusro/ckusro-core';
import { List } from '@material-ui/core';
import React from 'react';
import { useSelector } from 'react-redux';
import { State } from '../../../modules';
import FetchObjects from '../../FetchObject';
import TreeEntry from './TreeEntry';
import useTreeViewStyles from './useTreeViewStyles';
import { createRepositoriesManager } from '../../../models/RepositoriesManager';

type OwnProps = {
  internalPath: InternalPath;
};

type StateProps = {
  internalPath: InternalPath;
  treeEntries: TreeEntryType[];
};

export type TreeEntriesProps = StateProps;

export function TreeEntries({ treeEntries, internalPath }: TreeEntriesProps) {
  const entries = treeEntries.map((item) => (
    <TreeEntry
      key={item.oid + item.path}
      internalPath={createInternalPath(internalPath).join(item.path)}
      treeEntry={item}
    />
  ));
  const styles = useTreeViewStyles();

  return (
    <List
      dense={true}
      component="div"
      disablePadding
      className={styles.listStyle}
    >
      {entries}
    </List>
  );
}

export default function({ internalPath }: OwnProps) {
  const { treeEntries } = useSelector((state: State) => ({
    treeEntries: createRepositoriesManager(
      state.domain.repositories,
    ).fetchCurrentTreeEntries(internalPath),
  }));

  const stateProps: StateProps = {
    internalPath,
    treeEntries,
  };

  return (
    <FetchObjects oids={treeEntries.map(({ oid }) => oid)}>
      <TreeEntries {...stateProps} />
    </FetchObjects>
  );
}
