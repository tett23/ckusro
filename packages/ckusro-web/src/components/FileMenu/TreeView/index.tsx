import { RepoPath, url2RepoPath, RepositoryInfo } from '@ckusro/ckusro-core';
import { List, ListSubheader, Theme } from '@material-ui/core';
import { createStyles, makeStyles } from '@material-ui/styles';
import React from 'react';
import { useSelector } from 'react-redux';
import {
  createObjectManager,
  ObjectManager,
} from '../../../models/ObjectManager';
import { createRefManager, RefManager } from '../../../models/RefManager';
import { State } from '../../../modules';
import FetchObjects from '../../FetchObject';
import RepositoryComponent from './Repository';

type TreeViewStates = {
  repositories: RepositoryInfo[];
  objectManager: ObjectManager;
  refManager: RefManager;
};

export type TreeViewProps = TreeViewStates;

export function TreeView({
  repositories,
  objectManager,
  refManager,
}: TreeViewProps) {
  const classes = useStyles();

  const repos = repositories.map((item) => {
    const oid = createRefManager(refManager).headOid(url2RepoPath(
      item.url,
    ) as RepoPath);
    const commitObject =
      oid == null
        ? null
        : createObjectManager(objectManager).fetch(oid, 'commit');

    return (
      <FetchObjects key={item.url} oids={oid == null ? [] : [oid]}>
        <RepositoryComponent repository={item} commitObject={commitObject} />
      </FetchObjects>
    );
  });

  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      dense={true}
      subheader={
        <ListSubheader component="div" id="nested-list-subheader">
          Repositories
        </ListSubheader>
      }
      className={classes.root}
    >
      {repos}
    </List>
  );
}

export default function() {
  const state = useSelector(
    ({
      config: { repositories },
      domain: {
        repositories: { objectManager, refManager },
      },
    }: State) => {
      return { repositories, objectManager, refManager };
    },
  );

  return <TreeView {...state} />;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      maxWidth: 360,
      backgroundColor: theme.palette.background.paper,
    },
    nested: {
      paddingLeft: theme.spacing(4),
    },
  }),
);
