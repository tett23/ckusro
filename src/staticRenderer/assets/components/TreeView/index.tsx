import React from 'react';
import { CkusroFile } from '../../../../models/ckusroFile';
import { OutputContext } from '../../../../models/outputContext';
import buildNamespaceTree, { TreeViewItem } from './buildTree';
import TreeViewItemComponent from './TreeViewItem';

export type Props = {
  contexts: OutputContext[];
  files: CkusroFile[];
};

type Table = { [key in string]: CkusroFile };

export default function TreeView({ contexts, files }: Props) {
  const tv = buildNamespaceTree(files);
  const table = files.reduce(
    (acc, item) => {
      acc[item.id] = item;
      return acc;
    },
    {} as Table,
  );
  const items = tv.map((tvi) => {
    const context = contexts.find(
      (item) => item.name === table[tvi.id].namespace,
    );
    if (context == null) {
      throw new Error(`Context not found. name=${table[tvi.id].name}`);
    }

    return gen(table, context, tvi);
  });

  return <ul>{items}</ul>;
}

function gen(table: Table, context: OutputContext, tvi: TreeViewItem) {
  const file = table[tvi.id];
  if (file == null) {
    throw new Error('');
  }

  return (
    <TreeViewItemComponent key={file.id} context={context} file={file}>
      {tvi.children.map((item) => gen(table, context, item))}
    </TreeViewItemComponent>
  );
}
