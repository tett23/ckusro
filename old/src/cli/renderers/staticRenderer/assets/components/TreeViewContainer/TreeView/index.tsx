import React from 'react';
import { FileBuffer } from '../../../../../../../models/FileBuffer';
import { Leaf, TreeElement } from '../treeBuilder';
import TreeViewItem from './TreeViewItem';

export type Props = {
  nodes: TreeElement[];
  fileBuffers: FileBuffer[];
};

export default function TreeView({ nodes, fileBuffers }: Props) {
  const items = nodes.flatMap((node) => {
    const fileBuffer = fileBuffers.find((fb) => fb.id === node.id);
    if (fileBuffer == null) {
      return [];
    }

    if (isLeaf(node)) {
      return <TreeViewItem key={node.id} fileBuffer={fileBuffer} />;
    }

    return (
      <TreeView key={node.id} nodes={node.children} fileBuffers={fileBuffers} />
    );
  });

  return <ul>{items}</ul>;
}

function isLeaf(node: TreeElement): node is Leaf {
  return node.type === 'leaf';
}
