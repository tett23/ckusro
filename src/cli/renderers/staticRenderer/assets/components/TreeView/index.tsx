import React from 'react';
import {
  FileBuffer,
  fileBufferName,
} from '../../../../../../models/FileBuffer';
import { OutputContext } from '../../../../../../models/OutputContext';
import buildNamespaceTree, {
  TreeViewItem as TreeViewItemModel,
} from './buildTree';
import TreeViewItem from './TreeViewItem';

export type Props = {
  contexts: OutputContext[];
  files: FileBuffer[];
};

type Table = { [key in string]: FileBuffer };

export default function TreeView({ contexts, files }: Props) {
  const tv = buildNamespaceTree(files);
  const table = files.reduce(
    (acc, item) => {
      acc[item.id] = item;
      return acc;
    },
    {} as Table,
  );
  const items = tv.map((tvi) => {
    const context = contexts.find(
      (item) => item.name === table[tvi.id].namespace,
    );
    if (context == null) {
      throw new Error(
        `Context not found. name=${fileBufferName(table[tvi.id])}`,
      );
    }

    return gen(table, context, tvi);
  });

  return <ul>{items}</ul>;
}

function gen(table: Table, context: OutputContext, tvi: TreeViewItemModel) {
  const file = table[tvi.id];
  if (file == null) {
    throw new Error('');
  }

  return (
    <TreeViewItem key={file.id} context={context} file={file}>
      {tvi.children.map((item) => gen(table, context, item))}
    </TreeViewItem>
  );
}
