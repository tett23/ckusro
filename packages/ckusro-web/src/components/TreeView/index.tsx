import { CommitObject, RepoPath, url2RepoPath } from '@ckusro/ckusro-core';
import React from 'react';
import { useSelector } from 'react-redux';
import { createObjectManager, ObjectManager } from '../../models/ObjectManager';
import { createRefManager, RefManager } from '../../models/RefManager';
import { Repository } from '../../models/Repository';
import { State } from '../../modules';
import FetchObjects from '../FetchObject';
import { drawer } from '../shared';
import styled from '../styled';
import RepositoryComponent from './Repository';

type TreeViewStates = {
  repositories: Repository[];
  objectManager: ObjectManager;
  refManager: RefManager;
};

export type TreeViewProps = TreeViewStates;

export function TreeView({
  repositories,
  objectManager,
  refManager,
}: TreeViewProps) {
  const repos = repositories.map((item) => {
    const oid = createRefManager(refManager).headOid(url2RepoPath(
      item.url,
    ) as RepoPath);
    const commitObject =
      oid == null
        ? null
        : createObjectManager(objectManager).fetch<CommitObject>(oid);

    return (
      <FetchObjects key={item.url} oids={oid == null ? [] : [oid]}>
        <RepositoryComponent repository={item} commitObject={commitObject} />
      </FetchObjects>
    );
  });

  return <Wrapper>{repos}</Wrapper>;
}

const Wrapper = styled.View`
  ${drawer}
  overflow-y: scroll;
  overflow-x: hidden;
  padding: 0.25rem 0.5rem;
  width: 15vw;
  flex-basis: 15vw;
`;

export default function() {
  const state = useSelector(
    ({ domain: { repositories, objectManager, refManager } }: State) => {
      return { repositories, objectManager, refManager };
    },
  );

  return <TreeView {...state} />;
}
