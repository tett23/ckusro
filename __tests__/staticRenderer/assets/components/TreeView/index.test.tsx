import { shallow } from 'enzyme';
import React from 'react';
import TreeView from '../../../../../src/staticRenderer/assets/components/TreeView';
import { buildFile, buildLoaderContext } from '../../../../__fixtures__';

describe(TreeView, () => {
  it('renders correctly', () => {
    const contexts = [
      buildLoaderContext({ name: 'test1', path: '/test1' }),
      buildLoaderContext({ name: 'test2', path: '/test2' }),
    ];
    const files = [
      buildFile({ id: 'test1', namespace: 'test1', path: '/' }),
      buildFile({ id: 'test1', namespace: 'test1', path: '/foo.md' }),
      buildFile({ id: 'test2', namespace: 'test2', path: '/' }),
    ];
    const wrapper = shallow(<TreeView contexts={contexts} files={files} />);

    expect(wrapper.find('ul > TreeViewItem').length === 2);
    expect(wrapper).toMatchSnapshot();
  });

  it('throw Error when context does not exist', () => {
    const files = [buildFile({ id: 'test1' })];
    const wrapper = () => shallow(<TreeView contexts={[]} files={files} />);

    expect(wrapper).toThrowError();
  });
});
