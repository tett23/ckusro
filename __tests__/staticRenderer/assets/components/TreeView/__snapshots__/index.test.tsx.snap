// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TreeView
    contexts={
      Array [
        Object {
          "name": "test1",
          "path": "/test1",
        },
        Object {
          "name": "test2",
          "path": "/test2",
        },
      ]
    }
    files={
      Array [
        Object {
          "content": "[[bar.md]]",
          "fileType": "markdown",
          "id": "test1",
          "isLoaded": true,
          "name": "foo.md",
          "namespace": "test1",
          "path": "/",
          "strongDependencies": Array [
            "test:/bar.md",
          ],
          "variables": Array [],
          "weakDependencies": Array [
            "test:/bar.md",
          ],
        },
        Object {
          "content": "[[bar.md]]",
          "fileType": "markdown",
          "id": "test1",
          "isLoaded": true,
          "name": "foo.md",
          "namespace": "test1",
          "path": "/foo.md",
          "strongDependencies": Array [
            "test:/bar.md",
          ],
          "variables": Array [],
          "weakDependencies": Array [
            "test:/bar.md",
          ],
        },
        Object {
          "content": "[[bar.md]]",
          "fileType": "markdown",
          "id": "test2",
          "isLoaded": true,
          "name": "foo.md",
          "namespace": "test2",
          "path": "/",
          "strongDependencies": Array [
            "test:/bar.md",
          ],
          "variables": Array [],
          "weakDependencies": Array [
            "test:/bar.md",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TreeViewItem
          context={
            Object {
              "name": "test1",
              "path": "/test1",
            }
          }
          file={
            Object {
              "content": "[[bar.md]]",
              "fileType": "markdown",
              "id": "test1",
              "isLoaded": true,
              "name": "foo.md",
              "namespace": "test1",
              "path": "/foo.md",
              "strongDependencies": Array [
                "test:/bar.md",
              ],
              "variables": Array [],
              "weakDependencies": Array [
                "test:/bar.md",
              ],
            }
          }
        >
          <TreeViewItem
            context={
              Object {
                "name": "test1",
                "path": "/test1",
              }
            }
            file={
              Object {
                "content": "[[bar.md]]",
                "fileType": "markdown",
                "id": "test1",
                "isLoaded": true,
                "name": "foo.md",
                "namespace": "test1",
                "path": "/foo.md",
                "strongDependencies": Array [
                  "test:/bar.md",
                ],
                "variables": Array [],
                "weakDependencies": Array [
                  "test:/bar.md",
                ],
              }
            }
          />
        </TreeViewItem>,
        <TreeViewItem
          context={
            Object {
              "name": "test2",
              "path": "/test2",
            }
          }
          file={
            Object {
              "content": "[[bar.md]]",
              "fileType": "markdown",
              "id": "test2",
              "isLoaded": true,
              "name": "foo.md",
              "namespace": "test2",
              "path": "/",
              "strongDependencies": Array [
                "test:/bar.md",
              ],
              "variables": Array [],
              "weakDependencies": Array [
                "test:/bar.md",
              ],
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "test1",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TreeViewItem
              context={
                Object {
                  "name": "test1",
                  "path": "/test1",
                }
              }
              file={
                Object {
                  "content": "[[bar.md]]",
                  "fileType": "markdown",
                  "id": "test1",
                  "isLoaded": true,
                  "name": "foo.md",
                  "namespace": "test1",
                  "path": "/foo.md",
                  "strongDependencies": Array [
                    "test:/bar.md",
                  ],
                  "variables": Array [],
                  "weakDependencies": Array [
                    "test:/bar.md",
                  ],
                }
              }
            />,
          ],
          "context": Object {
            "name": "test1",
            "path": "/test1",
          },
          "file": Object {
            "content": "[[bar.md]]",
            "fileType": "markdown",
            "id": "test1",
            "isLoaded": true,
            "name": "foo.md",
            "namespace": "test1",
            "path": "/foo.md",
            "strongDependencies": Array [
              "test:/bar.md",
            ],
            "variables": Array [],
            "weakDependencies": Array [
              "test:/bar.md",
            ],
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "test1",
            "nodeType": "function",
            "props": Object {
              "children": Array [],
              "context": Object {
                "name": "test1",
                "path": "/test1",
              },
              "file": Object {
                "content": "[[bar.md]]",
                "fileType": "markdown",
                "id": "test1",
                "isLoaded": true,
                "name": "foo.md",
                "namespace": "test1",
                "path": "/foo.md",
                "strongDependencies": Array [
                  "test:/bar.md",
                ],
                "variables": Array [],
                "weakDependencies": Array [
                  "test:/bar.md",
                ],
              },
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "test2",
        "nodeType": "function",
        "props": Object {
          "children": Array [],
          "context": Object {
            "name": "test2",
            "path": "/test2",
          },
          "file": Object {
            "content": "[[bar.md]]",
            "fileType": "markdown",
            "id": "test2",
            "isLoaded": true,
            "name": "foo.md",
            "namespace": "test2",
            "path": "/",
            "strongDependencies": Array [
              "test:/bar.md",
            ],
            "variables": Array [],
            "weakDependencies": Array [
              "test:/bar.md",
            ],
          },
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TreeViewItem
            context={
              Object {
                "name": "test1",
                "path": "/test1",
              }
            }
            file={
              Object {
                "content": "[[bar.md]]",
                "fileType": "markdown",
                "id": "test1",
                "isLoaded": true,
                "name": "foo.md",
                "namespace": "test1",
                "path": "/foo.md",
                "strongDependencies": Array [
                  "test:/bar.md",
                ],
                "variables": Array [],
                "weakDependencies": Array [
                  "test:/bar.md",
                ],
              }
            }
          >
            <TreeViewItem
              context={
                Object {
                  "name": "test1",
                  "path": "/test1",
                }
              }
              file={
                Object {
                  "content": "[[bar.md]]",
                  "fileType": "markdown",
                  "id": "test1",
                  "isLoaded": true,
                  "name": "foo.md",
                  "namespace": "test1",
                  "path": "/foo.md",
                  "strongDependencies": Array [
                    "test:/bar.md",
                  ],
                  "variables": Array [],
                  "weakDependencies": Array [
                    "test:/bar.md",
                  ],
                }
              }
            />
          </TreeViewItem>,
          <TreeViewItem
            context={
              Object {
                "name": "test2",
                "path": "/test2",
              }
            }
            file={
              Object {
                "content": "[[bar.md]]",
                "fileType": "markdown",
                "id": "test2",
                "isLoaded": true,
                "name": "foo.md",
                "namespace": "test2",
                "path": "/",
                "strongDependencies": Array [
                  "test:/bar.md",
                ],
                "variables": Array [],
                "weakDependencies": Array [
                  "test:/bar.md",
                ],
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "test1",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TreeViewItem
                context={
                  Object {
                    "name": "test1",
                    "path": "/test1",
                  }
                }
                file={
                  Object {
                    "content": "[[bar.md]]",
                    "fileType": "markdown",
                    "id": "test1",
                    "isLoaded": true,
                    "name": "foo.md",
                    "namespace": "test1",
                    "path": "/foo.md",
                    "strongDependencies": Array [
                      "test:/bar.md",
                    ],
                    "variables": Array [],
                    "weakDependencies": Array [
                      "test:/bar.md",
                    ],
                  }
                }
              />,
            ],
            "context": Object {
              "name": "test1",
              "path": "/test1",
            },
            "file": Object {
              "content": "[[bar.md]]",
              "fileType": "markdown",
              "id": "test1",
              "isLoaded": true,
              "name": "foo.md",
              "namespace": "test1",
              "path": "/foo.md",
              "strongDependencies": Array [
                "test:/bar.md",
              ],
              "variables": Array [],
              "weakDependencies": Array [
                "test:/bar.md",
              ],
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "test1",
              "nodeType": "function",
              "props": Object {
                "children": Array [],
                "context": Object {
                  "name": "test1",
                  "path": "/test1",
                },
                "file": Object {
                  "content": "[[bar.md]]",
                  "fileType": "markdown",
                  "id": "test1",
                  "isLoaded": true,
                  "name": "foo.md",
                  "namespace": "test1",
                  "path": "/foo.md",
                  "strongDependencies": Array [
                    "test:/bar.md",
                  ],
                  "variables": Array [],
                  "weakDependencies": Array [
                    "test:/bar.md",
                  ],
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "test2",
          "nodeType": "function",
          "props": Object {
            "children": Array [],
            "context": Object {
              "name": "test2",
              "path": "/test2",
            },
            "file": Object {
              "content": "[[bar.md]]",
              "fileType": "markdown",
              "id": "test2",
              "isLoaded": true,
              "name": "foo.md",
              "namespace": "test2",
              "path": "/",
              "strongDependencies": Array [
                "test:/bar.md",
              ],
              "variables": Array [],
              "weakDependencies": Array [
                "test:/bar.md",
              ],
            },
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
